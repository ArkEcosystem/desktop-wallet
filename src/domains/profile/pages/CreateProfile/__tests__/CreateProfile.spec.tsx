import React from "react";
import { IntlProvider } from "react-intl";
import { fireEvent, render, act } from "@testing-library/react";

import { CreateProfile } from "../";

describe("CreateProfile", () => {
	it("should render", () => {
		const { container, asFragment, getByTestId, getAllByTestId } = render(
			<IntlProvider locale="en-US">
				<CreateProfile />
			</IntlProvider>,
		);

		expect(container).toBeTruthy();
		expect(getByTestId("list-divided__items")).toBeTruthy();
		expect(getByTestId("create-profile__form")).toBeTruthy();
		expect(getAllByTestId("list-divided-item__inner-wrapper").length).toEqual(1);
		expect(asFragment()).toMatchSnapshot();
	});
});
