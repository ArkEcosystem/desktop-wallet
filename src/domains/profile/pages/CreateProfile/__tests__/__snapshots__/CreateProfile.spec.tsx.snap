// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateProfile should render 1`] = `
<DocumentFragment>
  <div
    class="w-full h-full"
  >
    <nav
      class="bg-white shadow-header-smooth relative"
    >
      <div
        class="px-4 sm:px-6 lg:px-8"
      >
        <div
          class="relative flex justify-between h-20 md:h-24"
        >
          <div
            class="flex-shrink-0 flex items-center"
          >
            <div
              class="flex rounded-lg p-2 bg-logo"
            >
              <img
                alt="ARK Logo"
                class="h-6 md:h-8 lg:h-10"
                src="ark-logo.png"
              />
            </div>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="container mx-auto"
    >
      <div
        class="mx-auto my-8 md:w-3/4 lg:w-3/5 xl:w-1/2"
      >
        <h1
          class="mx-4 text-2xl font-bold md:text-4xl md:mx-8 xl:mx-16"
        >
          Create Profile
        </h1>
        <div
          class="mx-4 mt-2 text-theme-neutral-dark md:mx-8 xl:mx-16"
        >
          Create a new Profile or login with your MarketSquare account to get started.
        </div>
        <div
          class="mx-4 mt-5 md:mx-8 xl:mx-16"
        >
          <button
            class="sc-AxjAm gYunbo w-full"
            color="primary"
          >
            <div
              class="flex items-center justify-center"
            >
              <span>
                Login with MarketSquare
              </span>
            </div>
          </button>
          <div
            class="pt-4 mt-8 border-t border-theme-neutral-light"
          />
        </div>
      </div>
      <div
        class="mx-auto my-5 md:w-3/4 lg:w-3/5 xl:w-1/2"
      >
        <div
          class="mx-4 md:mx-8 xl:mx-16 mb-4 pb-5 border-b border-dashed border-theme-neutral-light"
        >
          <div
            class="flex flex-1"
          >
            <div
              class="w-full"
              data-testid="create-profile__form"
            >
              <ul
                data-testid="list-divided__items"
              >
                <li
                  class="flex flex-col w-full py-4 mb-5"
                  data-testid="list-divided-item__wrapper"
                >
                  <div
                    class="flex justify-between flex-col items-start"
                    data-testid="list-divided-item__inner-wrapper"
                  >
                    <div
                      class="flex flex-col"
                    >
                      <span
                        class="mr-5 undefined"
                        data-testid="list-divided-item__label"
                      >
                        New Profile
                      </span>
                      <span
                        class="text-sm text-theme-neutral text-sm font-semibold text-theme-neutral-dark"
                        data-testid="list-divided-item__label--description"
                      >
                        Select Profile Image
                      </span>
                    </div>
                  </div>
                  <div
                    class="mt-4"
                    data-testid="list-divided-item__content"
                  >
                    <div
                      class="flex flex-row mt-2"
                    >
                      <div
                        class="flex items-center justify-center w-24 h-24 mr-6 border-2 border-theme-neutral-light border-dashed rounded"
                      >
                        <button
                          class="flex items-center justify-center w-20 h-20 bg-theme-primary-contrast rounded-full"
                          type="button"
                        >
                          <div
                            class="sc-AxirZ eEbZjj"
                            height="20"
                            width="20"
                          >
                            <div>
                              <div>
                                <div
                                  data-src="upload.svg"
                                  src=""
                                />
                              </div>
                            </div>
                          </div>
                        </button>
                      </div>
                      <div
                        class="relative w-24 h-24 bg-theme-neutral-light rounded"
                      >
                        <img
                          alt="random avatar"
                          class="object-cover rounded"
                          src="https://randomuser.me/api/portraits/men/3.jpg"
                        />
                        <button
                          class="absolute flex items-center justify-center w-6 h-6 p-1 bg-theme-danger-contrast text-theme-danger rounded -top-3 -right-3"
                        >
                          <div
                            class="sc-AxirZ hDwaov"
                            height="12"
                            width="12"
                          >
                            <div>
                              <div>
                                <div
                                  data-src="close.svg"
                                  src=""
                                />
                              </div>
                            </div>
                          </div>
                        </button>
                      </div>
                    </div>
                  </div>
                </li>
              </ul>
              <form
                id="create-profile__form"
              >
                <div
                  class="flex flex-col"
                  data-testid="input__wrapper"
                >
                  <label
                    class="text-theme-medium"
                    data-testid="input__label"
                  >
                    Name
                    <div
                      class="mt-2 pb-2 flex relative items-center"
                    >
                      <input
                        class="form-input transition-colors duration-200 w-full"
                        name="name"
                        type="text"
                      />
                    </div>
                  </label>
                </div>
              </form>
            </div>
          </div>
        </div>
        <div
          class="mx-4 md:mx-8 xl:mx-16"
        >
          <button
            class="sc-AxjAm GWdpI"
            color="primary"
          >
            Back
          </button>
          <button
            class="sc-AxjAm gYunbo ml-2"
            color="primary"
            data-testid="create-profile__form--submit"
            form="create-profile__form"
            type="submit"
          >
            Complete
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
